### ISSUE TITLE
Implement violin charts

### ISSUE THREAD

**Issue by jrmoserbaltimore** [pgf-tikz/pgfplots#378]
Violin charts are the only type of chart capable of displaying complete distribution data over several cross-sectional data points in a time series, e.g. for displaying the distribution of wages, mean wage, and first/third quartiles per each year.

Useful approach in terms of `pgfplotstables`:

* A primary data file references other data files
* The primary data file columns are:
 * `data`:  the file (relative path `./foo.dat` accepted) containing all the plot data
 * `x`: the `x` axis label
 * `mean`:  The mean, calculated if not given or `{}`
 * `q1`:  First quartile, as mean.
 * `median`:  Median, as mean.
 * `q3`:  Third quartile, as mean.
 * `std`:  Standard deviation.
* The `data` field references a file containing points
 * if one column `value`, all single points
 * if two columns `count` and `value`, each point is duplicated `count` times

Note that `mean`, `median`, `q1`, and `q3` are given because while `pgfplots` may be able to compute small data sets, if you're using millions of data points you probably want to compute a few hundred or so representative data points and the mean, median, first and third quartiles, and standard deviation for plotting a decent-resolution chart.

Plots usually represent mean and standard deviation; median and inter-quartile range should also be an option.  That part is just a type of box-and-whiskers plotting.

**Comment by tapyu** [comment]
@michaeldorner How did you make this plot, I need to make violin plots on pgfplots.

**Comment by michaeldorner** [comment]
I attached the source. But it's not PGF since there is no support yet for them.

**Comment by tapyu** [comment]
@michaeldorner Yes, I had seen that you wrote in Python... I am afraid of this approach be poorly integrated with LaTeX :disappointed:

**Comment by tapyu** [comment]
@michaeldorner Now there is a solution! Check out my [answer](https://tex.stackexchange.com/a/679168/222785)!

**Comment by michaeldorner** [comment]
What an amazing work! Thanks for sharing!!! ❤️
